### PR TITLE
charles: 4.2.8 -> 4.6.1

### PR DESCRIPTION
**Pull request by ktor**
###### Motivation for this change
New Charles Proxy version. Tested both charles3 and charles4 to work.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/charles/default.nix [hunk 1]
```diff
@@ -1,13 +1,15 @@
 {
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/charles/default.nix:1]:
```suggestion
{ lib
```

File: pkgs/applications/networking/charles/default.nix [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
-lib, stdenv
+lib
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/charles/default.nix:2]:
```suggestion
```

File: pkgs/applications/networking/charles/default.nix [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
+, stdenv
 , makeWrapper
 , makeDesktopItem
 , fetchurl
 , jdk8
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/charles/default.nix:7]:
```suggestion
```

File: pkgs/applications/networking/charles/default.nix [hunk 4]
```diff
@@ -56,12 +58,16 @@ let
 
 in {
   charles4 = (generic {
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/charles/default.nix:60]: Can you extract this file into 3 files similar to generic.nix, 3.nix and 4.nix with the respective contents.

**Comment by FliegendeWurst** [comment]
This update was already done in #117939.

**Pull request closed without merging**